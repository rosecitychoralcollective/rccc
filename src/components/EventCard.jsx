import React from 'react';
import { makeStyles } from '@material-ui/styles';
import {
  Card, CardActionArea, CardMedia, CardContent, Typography, CardActions,
} from '@material-ui/core';
import translateString from '../utils/StringHelper';

const EventCard = ({ event, handleClick }) => {
  const t = translateString;

  const useStyles = makeStyles({
    card: {
      height: 'auto',
      maxHeight: 300,
      width: 'auto',
      maxWidth: 300,
      margin: '1rem',
      padding: '0.5rem',
      display: 'inline-block',
    },
  });

  const classes = useStyles();

  const maxDescLength = 75;
  const desc = event.description;
  let text = desc;
  if (desc.length > maxDescLength) {
    const mod = desc[maxDescLength - 1] === ' ' ? 1 : 0;
    text = `${desc.substring(0, maxDescLength - mod)}...`;
  }

  return (
    <Card className={classes.card} onClick={() => handleClick(event.id)}>
      <CardActionArea>
        <CardMedia
          component="img"
          alt={t('Event-Card-Image-Error')}
          height="auto"
          src={event.image}
          title="A Cat"
        />
        <CardContent>
          <Typography gutterBottom variant="h6">
            {event.title}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="h6">
            {text}
          </Typography>
          <Typography variant="subtitle2" color="textSecondary" component="p">
            {event.date}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

export default EventCard;
