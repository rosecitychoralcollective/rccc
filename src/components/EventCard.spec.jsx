import React from 'react';
import { shallow } from 'enzyme';
import Chance from 'chance';
import Resources from '../resources/resources';

import EventCard from './EventCard';

const chance = new Chance();

describe('Event Card Component defaults', () => {
  const event = {
    description: 'asdf',
    id: 0,
    title: 'soter',
    date: '10/31/20 7:00pm',
    image: Resources.placeholderCat,
  };
  let component;
  const handleClick = jest.fn();
  beforeAll(() => {
    component = shallow(<EventCard event={event} handleClick={handleClick} />);
  });
  it('Renders without crashing', () => {
    expect(component).toMatchSnapshot();
  });
  it('handles clicks', () => {
    component.find(`#card-${event.title}`).simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
  it('has a title', () => {
    expect(component.find('#card-title').text()).toBe(event.title);
  });
  it('has a description', () => {
    expect(component.find('#card-description').text()).toBe(event.description);
  });
  it('has a date', () => {
    expect(component.find('#card-date').text()).toBe(event.date);
  });
});

describe('Event card component boundaries', () => {
  const event = {
    description: chance.string({ length: 78 }),
    id: 0,
    title: 'soter',
    date: '10/31/20 7:00pm',
    image: Resources.placeholderCat,
  };
  it('truncates the description and adds an ellipsis', () => {
    const component = shallow(<EventCard event={event} />);
    const modifiedDescription = `${event.description.slice(0, 75)}...`;
    expect(component.find('#card-description').text()).toBe(modifiedDescription);
  });
  it('removes ending space if one is present before truncation', () => {
    event.description = `${event.description.slice(0, 74)} `;
    const component = shallow(<EventCard event={event} />);
    expect(component.find('#card-description').text()).toBe(event.description);
  });
});
