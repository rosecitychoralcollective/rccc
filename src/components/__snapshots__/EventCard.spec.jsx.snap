// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Card Component defaults Renders without crashing 1`] = `ShallowWrapper {}`;
